### ISSUE TITLE
cmd/go: go mod tidy -e should tolerate DNS resolution and network failures

### ISSUE THREAD

**Issue by oakad** [golang/go#53136]
go version go1.18.2 linux/amd64

`go mod tidy -e` should do whatever tidying it can while ignoring errors and it indeed does it quite well. Unfortunately, when some remote host can not be resolved (either directly or by module proxy), it fails hard, as following:

> go: finding module for package funny.host.com/some/unavailable/package
funny.host.com/some/other/package imports
        funny.host.com/some/unavailable/package: cannot find module providing package funny.host.com/some/unavailable/package: unrecognized import path "funny.host.com/some/unavailable/package": https fetch: Get "https://funny.host.com/some/unavailable/package?go-get=1": dial tcp: lookup funny.host.com: no such host

I believe that it should not fail hard, but rather treat the network/DNS error as "skippable", in line with all the other possible errors.

For example, in my particular case, the missing packages are not really missing - they belong to the same project, but need to be generated (things like protobuf schema and such). Yet, I cannot tidy the rest of the project without making the remote git repo somehow available.

**Comment by bcmills** [comment]
I agree that in principle the `-e` flag should push past these kinds of errors, although it might take a fair bit of debugging to figure out exactly where the error is not being propagated correctly.

**Comment by HowJMay** [comment]
let me try it!

**Comment by HowJMay** [comment]
Hi @oakad ! I have a question, after investigating the codebase. I found the exit code is 0 under the provided situation. May I ask the expected behavior in more detailed?

**Comment by oakad** [comment]
I have tested this myself and it's not quite conclusive; it is possible it was some sort of glitch on my side.

Or, may be the error which slips through is not of trivial nature, and happens somewhere between network failure, dns failure or sum db error response. But then, it may not be worth the time to hunt it down.

**Comment by bcmills** [comment]
Sounds like we need to figure out how to actually reproduce these errors. (If we can find a way to at least identify the root cause, then I agree that we should fix them, but I agree that it might not be worth spending a lot of time hunting it down.)
